File: measurecompute/DAQ/C#/ULDO01/ULDO01.cs
// ==============================================================================
//
//  File:                         ULDO01.CS
//
//  Library Call Demonstrated:    MccDaq.MccBoard.DOut()
//
//  Purpose:                      Writes a byte to digital output ports.
//
//  Demonstration:                Configures FirstPortA for output and writes
//                                a value to the port.
//
//  Other Library Calls:          MccDaq.MccBoard.DConfigPort()
//                                MccDaq.MccService.ErrHandling()
//
//  Special Requirements:         Board 0 must have a digital output port.
//                                Board 0 must have programmable digital ports.
//
// ==============================================================================
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using System.Diagnostics;

using MccDaq;

namespace ULDO01
{
	public class frmSetDigOut : System.Windows.Forms.Form
	{
		// Required by the Windows Form Designer
		private System.ComponentModel.IContainer components;

		public ToolTip ToolTip1;
		public Button cmdEndProgram;
		public TextBox txtValSet;
		public HScrollBar hsbSetDOutVal;
		public Label lblShowValOut;
		public Label lblDataValOut;
		public Label lblValSet;
		public Label lblInstruct;
		public Label lblDemoFunction;

		private MccDaq.MccBoard DaqBoard;
		const MccDaq.DigitalPortType PortNum = MccDaq.DigitalPortType.FirstPortA; //  set port to use
		const MccDaq.DigitalPortDirection Direction = MccDaq.DigitalPortDirection.DigitalOut; //  program digital port A for output

	 
		public frmSetDigOut()
		{
	        
			// This call is required by the Windows Form Designer.
			InitializeComponent();

			//  Initiate error handling
			//   activating error handling will trap errors like
			//   bad channel numbers and non-configured conditions.
			//   Parameters:
			//     MccDaq.ErrorReporting.PrintAll :all warnings and errors encountered will be printed
			//     MccDaq.ErrorHandling.StopAll   :if an error is encountered, the program will stop
			
			MccDaq.ErrorInfo ULStat = MccDaq.MccService.ErrHandling(MccDaq.ErrorReporting.PrintAll, MccDaq.ErrorHandling.StopAll);
			

			// Create a new MccBoard object for Board 0
			DaqBoard = new MccDaq.MccBoard(0);

			//  configure FirstPortA for digital output
			//   Parameters:
			//     PortNum    :the output port
			//     Direction  :sets the port for input or output
			ULStat = DaqBoard.DConfigPort(PortNum, Direction);
			
		}

		// Form overrides dispose to clean up the component list.
		protected override void  Dispose(bool Disposing)
		{
			if (Disposing)
			{
				if (components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose(Disposing);
		}
  
	    
		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
	    
		private void InitializeComponent()
		{
		this.components = new System.ComponentModel.Container();
		this.ToolTip1 = new System.Windows.Forms.ToolTip(this.components);
		this.cmdEndProgram = new System.Windows.Forms.Button();
		this.txtValSet = new System.Windows.Forms.TextBox();
		this.hsbSetDOutVal = new System.Windows.Forms.HScrollBar();
		this.lblShowValOut = new System.Windows.Forms.Label();
		this.lblDataValOut = new System.Windows.Forms.Label();
		this.lblValSet = new System.Windows.Forms.Label();
		this.lblInstruct = new System.Windows.Forms.Label();
		this.lblDemoFunction = new System.Windows.Forms.Label();
		this.SuspendLayout();
		// 
		// cmdEndProgram
		// 
		this.cmdEndProgram.BackColor = System.Drawing.SystemColors.Control;
		this.cmdEndProgram.Cursor = System.Windows.Forms.Cursors.Default;
		this.cmdEndProgram.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
		this.cmdEndProgram.ForeColor = System.Drawing.SystemColors.ControlText;
		this.cmdEndProgram.Location = new System.Drawing.Point(248, 216);
		this.cmdEndProgram.Name = "cmdEndProgram";
		this.cmdEndProgram.RightToLeft = System.Windows.Forms.RightToLeft.No;
		this.cmdEndProgram.Size = new System.Drawing.Size(57, 33);
		this.cmdEndProgram.TabIndex = 7;
		this.cmdEndProgram.Text = "Quit";
		this.cmdEndProgram.Click += new System.EventHandler(this.cmdEndProgram_Click);
		// 
		// txtValSet
		// 
		this.txtValSet.AcceptsReturn = true;
		this.txtValSet.AutoSize = false;
		this.txtValSet.BackColor = System.Drawing.SystemColors.Window;
		this.txtValSet.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle;
		this.txtValSet.Cursor = System.Windows.Forms.Cursors.IBeam;
		this.txtValSet.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
		this.txtValSet.ForeColor = System.Drawing.SystemColors.WindowText;
		this.txtValSet.Location = new System.Drawing.Point(264, 120);
		this.txtValSet.MaxLength = 0;
		this.txtValSet.Name = "txtValSet";
		this.txtValSet.RightToLeft = System.Windows.Forms.RightToLeft.No;
		this.txtValSet.Size = new System.Drawing.Size(41, 25);
		this.txtValSet.TabIndex = 4;
		this.txtValSet.Text = "0";
		this.txtValSet.TextChanged += new System.EventHandler(this.txtValSet_TextChanged);
		// 
		// hsbSetDOutVal
		// 
		this.hsbSetDOutVal.Cursor = System.Windows.Forms.Cursors.Default;
		this.hsbSetDOutVal.LargeChange = 51;
		this.hsbSetDOutVal.Location = new System.Drawing.Point(24, 120);
		this.hsbSetDOutVal.Maximum = 305;
		this.hsbSetDOutVal.Name = "hsbSetDOutVal";
		this.hsbSetDOutVal.RightToLeft = System.Windows.Forms.RightToLeft.No;
		this.hsbSetDOutVal.Size = new System.Drawing.Size(225, 17);
		this.hsbSetDOutVal.TabIndex = 1;
		this.hsbSetDOutVal.TabStop = true;
		this.hsbSetDOutVal.Scroll += new System.Windows.Forms.ScrollEventHandler(this.hsbSetDOutVal_Scroll);
		// 
		// lblShowValOut
		// 
		this.lblShowValOut.BackColor = System.Drawing.SystemColors.Window;
		this.lblShowValOut.Cursor = System.Windows.Forms.Cursors.Default;
		this.lblShowValOut.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
		this.lblShowValOut.ForeColor = System.Drawing.Color.Blue;
		this.lblShowValOut.Location = new System.Drawing.Point(224, 176);
		this.lblShowValOut.Name = "lblShowValOut";
		this.lblShowValOut.RightToLeft = System.Windows.Forms.RightToLeft.No;
		this.lblShowValOut.Size = new System.Drawing.Size(57, 17);
		this.lblShowValOut.TabIndex = 3;
		// 
		// lblDataValOut
		// 
		this.lblDataValOut.BackColor = System.Drawing.SystemColors.Window;
		this.lblDataValOut.Cursor = System.Windows.Forms.Cursors.Default;
		this.lblDataValOut.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
		this.lblDataValOut.ForeColor = System.Drawing.SystemColors.WindowText;
		this.lblDataValOut.Location = new System.Drawing.Point(32, 176);
		this.lblDataValOut.Name = "lblDataValOut";
		this.lblDataValOut.RightToLeft = System.Windows.Forms.RightToLeft.No;
		this.lblDataValOut.Size = new System.Drawing.Size(185, 17);
		this.lblDataValOut.TabIndex = 2;
		// 
		// lblValSet
		// 
		this.lblValSet.BackColor = System.Drawing.SystemColors.Window;
		this.lblValSet.Cursor = System.Windows.Forms.Cursors.Default;
		this.lblValSet.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
		this.lblValSet.ForeColor = System.Drawing.SystemColors.WindowText;
		this.lblValSet.Location = new System.Drawing.Point(256, 96);
		this.lblValSet.Name = "lblValSet";
		this.lblValSet.RightToLeft = System.Windows.Forms.RightToLeft.No;
		this.lblValSet.Size = new System.Drawing.Size(65, 17);
		this.lblValSet.TabIndex = 6;
		this.lblValSet.Text = "Value set:";
		// 
		// lblInstruct
		// 
		this.lblInstruct.BackColor = System.Drawing.SystemColors.Window;
		this.lblInstruct.Cursor = System.Windows.Forms.Cursors.Default;
		this.lblInstruct.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
		this.lblInstruct.ForeColor = System.Drawing.Color.Red;
		this.lblInstruct.Location = new System.Drawing.Point(56, 48);
		this.lblInstruct.Name = "lblInstruct";
		this.lblInstruct.RightToLeft = System.Windows.Forms.RightToLeft.No;
		this.lblInstruct.Size = new System.Drawing.Size(201, 25);
		this.lblInstruct.TabIndex = 5;
		this.lblInstruct.Text = "Set output value using scroll bar or enter value in Value Set box:";
		this.lblInstruct.TextAlign = System.Drawing.ContentAlignment.TopCenter;
		// 
		// lblDemoFunction
		// 
		this.lblDemoFunction.BackColor = System.Drawing.SystemColors.Window;
		this.lblDemoFunction.Cursor = System.Windows.Forms.Cursors.Default;
		this.lblDemoFunction.Font = new System.Drawing.Font("Arial", 8F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
		this.lblDemoFunction.ForeColor = System.Drawing.SystemColors.WindowText;
		this.lblDemoFunction.Location = new System.Drawing.Point(24, 8);
		this.lblDemoFunction.Name = "lblDemoFunction";
		this.lblDemoFunction.RightToLeft = System.Windows.Forms.RightToLeft.No;
		this.lblDemoFunction.Size = new System.Drawing.Size(289, 25);
		this.lblDemoFunction.TabIndex = 0;
		this.lblDemoFunction.Text = "Demonstration of MccDaq.MccBoard.DOut()";
		this.lblDemoFunction.TextAlign = System.Drawing.ContentAlignment.TopCenter;
		// 
		// frmSetDigOut
		// 
		this.AutoScaleBaseSize = new System.Drawing.Size(6, 13);
		this.BackColor = System.Drawing.SystemColors.Window;
		this.ClientSize = new System.Drawing.Size(331, 267);
		this.Controls.AddRange(new System.Windows.Forms.Control[] {
																		this.cmdEndProgram,
																		this.txtValSet,
																		this.hsbSetDOutVal,
																		this.lblShowValOut,
																		this.lblDataValOut,
																		this.lblValSet,
																		this.lblInstruct,
																		this.lblDemoFunction});
		this.Font = new System.Drawing.Font("Arial", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
		this.ForeColor = System.Drawing.SystemColors.WindowText;
		this.Location = new System.Drawing.Point(7, 103);
		this.Name = "frmSetDigOut";
		this.StartPosition = System.Windows.Forms.FormStartPosition.Manual;
		this.Text = "Universal Library Digital Output";
		this.ResumeLayout(false);

		}

	#endregion

		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new frmSetDigOut());
		}
	    

		private void cmdEndProgram_Click(object eventSender, System.EventArgs eventArgs) /* Handles cmdEndProgram.Click */
		{
			ushort DataValue=0;
			MccDaq.ErrorInfo ULStat = DaqBoard.DOut(PortNum, DataValue);
			

			Application.Exit();
		}


		private void hsbSetDOutVal_Change(int newScrollValue)
		{
			//  get a value to write to the port
			ushort DataValue = (ushort)newScrollValue;
			txtValSet.Text = DataValue.ToString("0");

			//  write the value to FirstPortA
			//   Parameters:
			//     PortNum    :the output port
			//     DataValue  :the value written to the port
			MccDaq.ErrorInfo ULStat = DaqBoard.DOut(PortNum, DataValue);

			if (ULStat.Value == MccDaq.ErrorInfo.ErrorCode.NoErrors)
			{
				lblDataValOut.Text = "Value written to FirstPortA:";
				lblShowValOut.Text = DataValue.ToString("0");
			}

		}


		private void txtValSet_TextChanged(object eventSender, System.EventArgs eventArgs) /* Handles txtValSet.TextChanged */
		{
			if (Convert.ToInt16(txtValSet.Text) > 255) 
				txtValSet.Text = "255";

			hsbSetDOutVal.Value = int.Parse(txtValSet.Text);
			hsbSetDOutVal_Change(hsbSetDOutVal.Value);
		}

		private void hsbSetDOutVal_Scroll(object eventSender, System.Windows.Forms.ScrollEventArgs eventArgs) /* Handles hsbSetDOutVal.Scroll */
		{
			if (eventArgs.Type == System.Windows.Forms.ScrollEventType.EndScroll)
				hsbSetDOutVal_Change(eventArgs.NewValue);
		}

	}
}

File: measurecompute/DAQ/VB.NET/ULDO01.NET/ULDO01.vb
'==============================================================================

' File:                         ULDO01.VB

' Library Call Demonstrated:    MccDaq.MccBoard.DOut()

' Purpose:                      Writes a byte to digital output ports.

' Demonstration:                Configures FirstPortA for output and writes
'                               a value to the port.

' Other Library Calls:          MccDaq.MccBoard.DConfigPort()
'                               MccDaq.MccService.ErrHandling()

' Special Requirements:         Board 0 must have a digital output port.
'                               Board 0 must have programmable digital ports.

'==============================================================================
Option Strict Off
Option Explicit On 
Public Class frmSetDigOut
    Inherits System.Windows.Forms.Form

    Const PortNum As MccDaq.DigitalPortType = MccDaq.DigitalPortType.FirstPortA ' set port to use
    Const Direction As MccDaq.DigitalPortDirection = MccDaq.DigitalPortDirection.DigitalOut ' program digital port A for output

    'Create a new MccBoard object for Board 0 
    Private DaqBoard As MccDaq.MccBoard = New MccDaq.MccBoard(0)

    Private Sub cmdEndProgram_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdEndProgram.Click
        Dim ULStat As MccDaq.ErrorInfo
        Dim DataValue As UInt16

        DataValue = Convert.ToUInt16(0)

        ULStat = DaqBoard.DOut(PortNum, DataValue)
        If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then Stop

        End

    End Sub

    Private Sub hsbSetDOutVal_Change(ByVal newScrollValue As Integer)
        Dim ULStat As MccDaq.ErrorInfo
        Dim DataValue As UInt16

        ' get a value to write to the port

        DataValue = Convert.ToUInt16(newScrollValue)
        txtValSet.Text = DataValue.ToString("0")

        ' write the value to FirstPortA
        '  Parameters:
        '    PortNum    :the output port
        '    DataValue  :the value written to the port

        ULStat = DaqBoard.DOut(PortNum, DataValue)

        If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then
            Stop
        Else
            lblDataValOut.Text = "Value written to FirstPortA:"
            lblShowValOut.Text = DataValue.ToString("0")
        End If

    End Sub

    Private Sub txtValSet_TextChanged(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtValSet.TextChanged

        If Val(txtValSet.Text) > 255 Then txtValSet.Text = "255"
        hsbSetDOutVal.Value = Integer.Parse(txtValSet.Text)

    End Sub
    Private Sub hsbSetDOutVal_Scroll(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.ScrollEventArgs) Handles hsbSetDOutVal.Scroll
        Select Case eventArgs.Type
            Case System.Windows.Forms.ScrollEventType.EndScroll
                hsbSetDOutVal_Change(eventArgs.NewValue)
        End Select
    End Sub
#Region "Windows Form Designer generated code "
    Public Sub New()
        MyBase.New()

        'This call is required by the Windows Form Designer.
        InitializeComponent()

        InitUL()

    End Sub
    'Form overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal Disposing As Boolean)
        If Disposing Then
            If Not components Is Nothing Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(Disposing)
    End Sub
    'Required by the Windows Form Designer
    Private components As System.ComponentModel.IContainer
    Public ToolTip1 As System.Windows.Forms.ToolTip
    Public WithEvents cmdEndProgram As System.Windows.Forms.Button
    Public WithEvents txtValSet As System.Windows.Forms.TextBox
    Public WithEvents hsbSetDOutVal As System.Windows.Forms.HScrollBar
    Public WithEvents lblShowValOut As System.Windows.Forms.Label
    Public WithEvents lblDataValOut As System.Windows.Forms.Label
    Public WithEvents lblValSet As System.Windows.Forms.Label
    Public WithEvents lblInstruct As System.Windows.Forms.Label
    Public WithEvents lblDemoFunction As System.Windows.Forms.Label
    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.
    'Do not modify it using the code editor.
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Dim resources As System.Resources.ResourceManager = New System.Resources.ResourceManager(GetType(frmSetDigOut))
        Me.components = New System.ComponentModel.Container()
        Me.ToolTip1 = New System.Windows.Forms.ToolTip(components)
        Me.ToolTip1.Active = True
        Me.cmdEndProgram = New System.Windows.Forms.Button()
        Me.txtValSet = New System.Windows.Forms.TextBox()
        Me.hsbSetDOutVal = New System.Windows.Forms.HScrollBar()
        Me.lblShowValOut = New System.Windows.Forms.Label()
        Me.lblDataValOut = New System.Windows.Forms.Label()
        Me.lblValSet = New System.Windows.Forms.Label()
        Me.lblInstruct = New System.Windows.Forms.Label()
        Me.lblDemoFunction = New System.Windows.Forms.Label()
        Me.StartPosition = System.Windows.Forms.FormStartPosition.Manual
        Me.BackColor = System.Drawing.SystemColors.Window
        Me.Text = "Universal Library Digital Output"
        Me.ClientSize = New System.Drawing.Size(331, 267)
        Me.Location = New System.Drawing.Point(7, 103)
        Me.Font = New System.Drawing.Font("Arial", 8.25!, System.Drawing.FontStyle.Bold Or System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.ForeColor = System.Drawing.SystemColors.WindowText
        Me.AutoScaleBaseSize = New System.Drawing.Size(6, 13)
        Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.Sizable
        Me.ControlBox = True
        Me.Enabled = True
        Me.KeyPreview = False
        Me.MaximizeBox = True
        Me.MinimizeBox = True
        Me.Cursor = System.Windows.Forms.Cursors.Default
        Me.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.ShowInTaskbar = True
        Me.HelpButton = False
        Me.WindowState = System.Windows.Forms.FormWindowState.Normal
        Me.Name = "frmSetDigOut"
        Me.cmdEndProgram.TextAlign = System.Drawing.ContentAlignment.MiddleCenter
        Me.cmdEndProgram.Text = "Quit"
        Me.cmdEndProgram.Size = New System.Drawing.Size(57, 33)
        Me.cmdEndProgram.Location = New System.Drawing.Point(248, 216)
        Me.cmdEndProgram.TabIndex = 7
        Me.cmdEndProgram.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.cmdEndProgram.BackColor = System.Drawing.SystemColors.Control
        Me.cmdEndProgram.CausesValidation = True
        Me.cmdEndProgram.Enabled = True
        Me.cmdEndProgram.ForeColor = System.Drawing.SystemColors.ControlText
        Me.cmdEndProgram.Cursor = System.Windows.Forms.Cursors.Default
        Me.cmdEndProgram.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.cmdEndProgram.TabStop = True
        Me.cmdEndProgram.Name = "cmdEndProgram"
        Me.txtValSet.AutoSize = False
        Me.txtValSet.Size = New System.Drawing.Size(41, 25)
        Me.txtValSet.Location = New System.Drawing.Point(264, 120)
        Me.txtValSet.TabIndex = 4
        Me.txtValSet.Text = "0"
        Me.txtValSet.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.txtValSet.AcceptsReturn = True
        Me.txtValSet.TextAlign = System.Windows.Forms.HorizontalAlignment.Left
        Me.txtValSet.BackColor = System.Drawing.SystemColors.Window
        Me.txtValSet.CausesValidation = True
        Me.txtValSet.Enabled = True
        Me.txtValSet.ForeColor = System.Drawing.SystemColors.WindowText
        Me.txtValSet.HideSelection = True
        Me.txtValSet.ReadOnly = False
        Me.txtValSet.MaxLength = 0
        Me.txtValSet.Cursor = System.Windows.Forms.Cursors.IBeam
        Me.txtValSet.Multiline = False
        Me.txtValSet.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.txtValSet.ScrollBars = System.Windows.Forms.ScrollBars.None
        Me.txtValSet.TabStop = True
        Me.txtValSet.Visible = True
        Me.txtValSet.BorderStyle = System.Windows.Forms.BorderStyle.FixedSingle
        Me.txtValSet.Name = "txtValSet"
        Me.hsbSetDOutVal.Size = New System.Drawing.Size(225, 17)
        Me.hsbSetDOutVal.LargeChange = 51
        Me.hsbSetDOutVal.Location = New System.Drawing.Point(24, 120)
        Me.hsbSetDOutVal.Maximum = 305
        Me.hsbSetDOutVal.TabIndex = 1
        Me.hsbSetDOutVal.CausesValidation = True
        Me.hsbSetDOutVal.Enabled = True
        Me.hsbSetDOutVal.Minimum = 0
        Me.hsbSetDOutVal.Cursor = System.Windows.Forms.Cursors.Default
        Me.hsbSetDOutVal.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.hsbSetDOutVal.SmallChange = 1
        Me.hsbSetDOutVal.TabStop = True
        Me.hsbSetDOutVal.Value = 0
        Me.hsbSetDOutVal.Visible = True
        Me.hsbSetDOutVal.Name = "hsbSetDOutVal"
        Me.lblShowValOut.BackColor = System.Drawing.SystemColors.Window
        Me.lblShowValOut.ForeColor = System.Drawing.Color.Blue
        Me.lblShowValOut.Size = New System.Drawing.Size(57, 17)
        Me.lblShowValOut.Location = New System.Drawing.Point(224, 176)
        Me.lblShowValOut.TabIndex = 3
        Me.lblShowValOut.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblShowValOut.TextAlign = System.Drawing.ContentAlignment.TopLeft
        Me.lblShowValOut.Enabled = True
        Me.lblShowValOut.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblShowValOut.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblShowValOut.UseMnemonic = True
        Me.lblShowValOut.Visible = True
        Me.lblShowValOut.AutoSize = False
        Me.lblShowValOut.BorderStyle = System.Windows.Forms.BorderStyle.None
        Me.lblShowValOut.Name = "lblShowValOut"
        Me.lblDataValOut.BackColor = System.Drawing.SystemColors.Window
        Me.lblDataValOut.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblDataValOut.Size = New System.Drawing.Size(185, 17)
        Me.lblDataValOut.Location = New System.Drawing.Point(32, 176)
        Me.lblDataValOut.TabIndex = 2
        Me.lblDataValOut.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblDataValOut.TextAlign = System.Drawing.ContentAlignment.TopLeft
        Me.lblDataValOut.Enabled = True
        Me.lblDataValOut.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblDataValOut.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblDataValOut.UseMnemonic = True
        Me.lblDataValOut.Visible = True
        Me.lblDataValOut.AutoSize = False
        Me.lblDataValOut.BorderStyle = System.Windows.Forms.BorderStyle.None
        Me.lblDataValOut.Name = "lblDataValOut"
        Me.lblValSet.BackColor = System.Drawing.SystemColors.Window
        Me.lblValSet.Text = "Value set:"
        Me.lblValSet.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblValSet.Size = New System.Drawing.Size(65, 17)
        Me.lblValSet.Location = New System.Drawing.Point(256, 96)
        Me.lblValSet.TabIndex = 6
        Me.lblValSet.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblValSet.TextAlign = System.Drawing.ContentAlignment.TopLeft
        Me.lblValSet.Enabled = True
        Me.lblValSet.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblValSet.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblValSet.UseMnemonic = True
        Me.lblValSet.Visible = True
        Me.lblValSet.AutoSize = False
        Me.lblValSet.BorderStyle = System.Windows.Forms.BorderStyle.None
        Me.lblValSet.Name = "lblValSet"
        Me.lblInstruct.TextAlign = System.Drawing.ContentAlignment.TopCenter
        Me.lblInstruct.BackColor = System.Drawing.SystemColors.Window
        Me.lblInstruct.Text = "Set output value using scroll bar or enter value in Value Set box:"
        Me.lblInstruct.ForeColor = System.Drawing.Color.Red
        Me.lblInstruct.Size = New System.Drawing.Size(201, 25)
        Me.lblInstruct.Location = New System.Drawing.Point(56, 48)
        Me.lblInstruct.TabIndex = 5
        Me.lblInstruct.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblInstruct.Enabled = True
        Me.lblInstruct.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblInstruct.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblInstruct.UseMnemonic = True
        Me.lblInstruct.Visible = True
        Me.lblInstruct.AutoSize = False
        Me.lblInstruct.BorderStyle = System.Windows.Forms.BorderStyle.None
        Me.lblInstruct.Name = "lblInstruct"
        Me.lblDemoFunction.TextAlign = System.Drawing.ContentAlignment.TopCenter
        Me.lblDemoFunction.BackColor = System.Drawing.SystemColors.Window
        Me.lblDemoFunction.Text = "Demonstration of MccDaq.MccBoard.DOut()"
        Me.lblDemoFunction.ForeColor = System.Drawing.SystemColors.WindowText
        Me.lblDemoFunction.Size = New System.Drawing.Size(289, 25)
        Me.lblDemoFunction.Location = New System.Drawing.Point(24, 8)
        Me.lblDemoFunction.TabIndex = 0
        Me.lblDemoFunction.Font = New System.Drawing.Font("Arial", 8.0!, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
        Me.lblDemoFunction.Enabled = True
        Me.lblDemoFunction.Cursor = System.Windows.Forms.Cursors.Default
        Me.lblDemoFunction.RightToLeft = System.Windows.Forms.RightToLeft.No
        Me.lblDemoFunction.UseMnemonic = True
        Me.lblDemoFunction.Visible = True
        Me.lblDemoFunction.AutoSize = False
        Me.lblDemoFunction.BorderStyle = System.Windows.Forms.BorderStyle.None
        Me.lblDemoFunction.Name = "lblDemoFunction"
        Me.Controls.Add(cmdEndProgram)
        Me.Controls.Add(txtValSet)
        Me.Controls.Add(hsbSetDOutVal)
        Me.Controls.Add(lblShowValOut)
        Me.Controls.Add(lblDataValOut)
        Me.Controls.Add(lblValSet)
        Me.Controls.Add(lblInstruct)
        Me.Controls.Add(lblDemoFunction)
    End Sub
#End Region

#Region "Universal Library Initialization - Expand this region to change error handling, etc."

    Private Sub InitUL()
        Dim ULStat As MccDaq.ErrorInfo

        ' declare revision level of Universal Library

        ULStat = MccDaq.MccService.DeclareRevision(MccDaq.MccService.CurrentRevNum)

        ' Initiate error handling
        '  activating error handling will trap errors like
        '  bad channel numbers and non-configured conditions.
        '  Parameters:
        '    MccDaq.ErrorReporting.PrintAll :all warnings and errors encountered will be printed
        '    MccDaq.ErrorHandling.StopAll   :if any error is encountered, the program will stop


        ULStat = MccDaq.MccService.ErrHandling(MccDaq.ErrorReporting.PrintAll, MccDaq.ErrorHandling.StopAll)
        If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then
            Stop
        End If


        ' configure FirstPortA for digital output
        '  Parameters:
        '    PortNum    :the output port
        '    Direction  :sets the port for input or output

        ULStat = DaqBoard.DConfigPort(PortNum, Direction)
        If ULStat.Value <> MccDaq.ErrorInfo.ErrorCode.NoErrors Then Stop

    End Sub
#End Region

End Class